File: app/src/main/java/org/exallium/tradetracker/app/model/entities/Record.java
package org.exallium.tradetracker.app.model.entities;

import com.orm.SugarRecord;
import org.exallium.tradetracker.app.MainApplication;

public class Record<T> extends SugarRecord<T> {

    @Override
    public void save() {
        super.save();
        MainApplication.getInstance().onObjectSaved(this);
    }
}

File: app/src/main/kotlin/org/exallium/tradetracker/app/model/entities/Record.kt
package org.exallium.tradetracker.app.model.entities

import com.orm.SugarRecord
import org.exallium.tradetracker.app.MainApplication

public open class Record<T> : SugarRecord<T>() {

    override fun save() {
        super.save()
        // We should move to an Observer Model, where when we save a new Model, we emit it via an observer, and
        // it gets sent down the chain.  The Adapter sees this new item, finds the appropriate position in the view
        // to add it, and notifyAdded at the appropriate location, or notifyModified if this is just a modification
        // (The item already exists in the list)
        MainApplication.onObjectSavedSubject.onNext(this)
    }
}
